gobblin-metastore/src/main/java/gobblin/metastore/StateStore.java

```diff
@@ -34,6 +36,10 @@
  */
 public interface StateStore<T extends State> {
 
+  interface Factory {
+    <T extends State> StateStore<T> createStateStore(Properties props, Class<T> stateClass);
```

Comment:
I know I've missed this in the previous commit (for DatasetStateStore.Factory), but is it too late to switch to typesafe Config instead of Properties? Overall, we've been trying to move away from the latter and to the former  .